### PR TITLE
163 reusable platforms instruments

### PR DESCRIPTION
**Pull request by BryceMarshall**
Added persistent storage for instruments per user 
Added navbar tab to manage user instruments
updated UI on Instrument/Platform tab to reflect changes

### CODE REVIEW

**Comment by fostermh** [comment]
- when trying to create a platform in the platform menu from the left I can not enter a value into the platform type or platform id fields.
- TODO: now that there can be multiple platforms we need to link instruments to platforms

**Comment by BryceMarshall** [comment]
Merge conflicts with translation code resolved & basic functionality for platform and instruments verified.

## Bugs
- Duplicating instruments holds reference to
original
- Platform fields are not activating Required Mark
properly

**Comment by fostermh** [comment]
> * Duplicating instruments holds reference to original - saving either entry will modify the same underlying Firebase record

I suspect a 'deepCopy' would help you out here.

**Comment by BryceMarshall** [comment]
## Bug: Duplicate button
- Copy-paste errors:
	- Code existed to append a "(Copy)" string to make each entry unique, but it referenced fields from the Contact rather than the Instrument.
	- Updating the fields didn't resolve the issue.
### To Reproduce
- Add a saved instrument
- Duplicate it
- Save the new entry
- **Expected**
	- Two entries in saved dropdown
- **Actual**
	- New entry replaced the old one
### Resolution:
- Save Update Instrument method was referencing an identifier that had been renamed.
## Takeaways
- The 'Left List' pattern is a good candidate for code refactor- lots of duplication present.

**Comment by BryceMarshall** [comment]
## Bug: Platform Validator 
- Updated the Required Mark on the platform page to check the presence of each field rather than using the overall platform validator
- Updated the overall platform Validator to treat platforms as a list

File: src/components/FormComponents/Instruments.jsx [hunk 1]
```diff
@@ -54,55 +39,24 @@ const Instruments = ({
...
-          </Grid>
-        </Grid>
-      </Grid>
+              userInstruments={userInstruments}
+              saveUpdateInstrument={saveUpdateInstrument}
+              />
+    </Grid>
       <Grid item xs>
         <Grid container direction="column">
           {instrument && (
```
Comment (fostermh) [inline src/components/FormComponents/Instruments.jsx:62]: is this code duplicated in instrumentEditor.js?

File: src/components/FormComponents/Platform.jsx [hunk 2]
```diff
@@ -1,122 +1,159 @@
...
-          )}
-          options={platformsSorted.map((e) => e.label_en)}
-          disabled={disabled}
-          label={<I18n en="Platform" fr="Plateforme" />}
-          fullWidth={false}
-        />
-      </Grid>
+      <Grid item xs>
+        <Grid container direction="column">
+        { platform && (
```
Comment (fostermh) [inline src/components/FormComponents/Platform.jsx:59]: is this code duplicated in PlatformEditor.jsx?

**Comment by fostermh** [comment]
when selecting a platform that an instrument is part of. the label shows up as 'New platform' even though it has a name in the platform list.

Screenshot 2024-04-17 at 10 45 47 AM

**Review by fostermh — APPROVED** [review]

**Pull request merged**